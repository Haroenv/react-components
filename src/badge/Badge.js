import React from 'react';
import cx from 'classnames';
import { pure, skinnable, props, t } from '../utils';
import FlexView from '../flex/FlexView';

/** A small descriptive element to attach to other UI components
 * @param label - the descriptive content of the badge
 * @param active - tells if the badge is active (for styling purposes)
 */
@pure
@skinnable()
@props({
  label: t.ReactChildren,
  active: t.maybe(t.Boolean),
  className: t.maybe(t.String),
  style: t.maybe(t.Object)
})
export default class Badge extends React.Component {

  getLocals() {
    const { label, active, className, style } =  this.props;

    return {
      label,
      style,
      className: cx('badge', { active }, className)
    };
  }

  template({ label, className, style }) {
    return (
      <FlexView vAlignContent='center' hAlignContent='center' {...{ className, style }}>
        <span className='badge-label'>{label}</span>
      </FlexView>
    );
  }
}
