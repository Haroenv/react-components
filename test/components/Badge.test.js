import React from 'react';
import render from '../render';

import Badge from '../../src/badge';

describe('Badge', () => {

  it('renders correctly', () => {
    const tree = render(
      <Badge label='42' />
    );
    expect(tree).toMatchSnapshot();
  });

  describe('getLocals', () => {

    it('computes className', () => {
      const badge = new Badge({
        label: 42,
        className: 'awesome-class'
      });
      const { className } = badge.getLocals();
      expect(className).toContain('badge');
      expect(className).toContain('awesome-class');
      expect(className).not.toContain('active');
    });

    it('computes className when active', () => {
      const badge = new Badge({
        active: true,
        label: 42,
        className: 'awesome-class'
      });
      const { className } = badge.getLocals();
      expect(className).toContain('badge');
      expect(className).toContain('awesome-class');
      expect(className).toContain('active');
    });

  });

});
