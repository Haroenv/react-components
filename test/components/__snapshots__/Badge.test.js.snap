exports[`Badge renders correctly 1`] = `
"<FlexView vAlignContent=\"center\" hAlignContent=\"center\" className=\"badge\" style={[undefined]}>
<span className=\"badge-label\">
42
</span>
</FlexView>"
`;
